### DISCUSSION TITLE
Basic question: Output naming

### DISCUSSION THREAD

**Discussion by MacSass** [mifi/lossless-cut#2108] [General]
Hi,
I just found lossles-cut and it looks great to me.
I´m just struggling with getting the output file names as I would like to. I even read about regular expressions now being supported - but I can´t find any explanation where to define the template for the output filename.
I would have expected it in the settings dialogue in the "Options affecting exported files", but there is only an option to define the working directory.
Just before export I get a dialogue to set the output file name - but I can´t find where to define the template that this is using ...

Sorry if that is a basic question - I could not find it in the help file documentation and not here in the discussions ...

Thanks!

**Comment by mifi** [comment]
Hey, no worries. Once you open the export dialog, make sure "Export mode" is set to "separate files", then beloq "Output file names:", click on the name and the template editor will pop out:

<img width="679" alt="Screenshot 2024-08-13 at 17 21 51" src="https://github.com/user-attachments/assets/527e4a14-e158-4bd0-94d1-769a53e48f87">

**Comment by MacSass** [comment]
Ahh - great. In my case I want to merge the cuts - so I set the mode to "Merge cuts", but in fact I really must have missed the template editor.
Maybe if not to cumbersome it might make sense to replicate it in the settings dialogue - I guess this is where newbies like me would look for it. The current place seems logical but it is not obvious, even when clicking on the name you might miss it (I think I did :-) ).

Maybe this even opens the option to have more than one template that you can switch between - but this might be overkill.

Thanks for your response and nice work!

PS: I have one more request, but will open a new thread for it, to keep conversation clean ...

**Reply by mifi** [reply]
i will make it more clear that you can click the file name and also a helping text reference from settings

**Reply by MacSass** [reply]
Thanks - sounds good. I guess if I made the mistake, others might too and ask themselves the same question :-)

**Reply by MacSass** [reply]
@mifi : I need to come back to this topic once more, as I was just working on some files and the output file name editor did not come up.
So I was a bit puzzled, as I had it after your last comment ... I looked at your comment again and saw the remark about "Separate files".
In fact the output file name editor only comes up with "separate files" not with "merge cuts", this is why I had never seen it before ...
From my point of view having a output file name template also makes sense for "merge cuts", as I want my resulting file named as "{filename}-cut{ext}" always. 

![grafik](https://github.com/user-attachments/assets/5463e2a6-7f01-4957-94c2-3de0e4bae904)

From my point of view the template editor is great - but user guidance how to use it can be enhanced - especially for novice users.
It would be nice to be able to save a template and have several once you can select from drop-down.

Personally I would move the output file name template editor to the settings dialog fully, allowing for several templates.
Then, in the export settings dialog just have the drop down to select the desired template. I think this is the most common behavior of most apps (e.g. export in Adobe Lightroom).

Let me know what you think - at least I´m getting used to LosslessCut :-)

**Reply by mifi** [reply]
I agree it makes sense to be able to use the template editor for merged output too. will implement that. saving a template would be nice, but i think it's not so high priority because as a workaround people can just copy-paste from a text document to achieve the same

**Reply by MacSass** [reply]
Yes - think you are right, being able to store different presets might be "overkill" ....

**Reply by NickFromTheNetherlands** [reply]
That is very strange that the outputfile naming ${FILENAME}_1${EXT} is not available in merge????????????

**Reply by mifi** [reply]
@NickFromTheNetherlands try with the latest nightly build

**Reply by NickFromTheNetherlands** [reply]
V42 beta will show a pencil. Thanks.
![modify filename with export merge cut is  working after click on the pencil (beta V42)](https://github.com/user-attachments/assets/c65e5530-92df-4363-805a-0fa5f38a3389)
